crates/resolver/src/name_server/name_server_pool.rs

```diff
@@ -258,29 +258,30 @@ where
             debug!("sending request: {:?}", request.queries());
 
             // First try the UDP connections
-            let udp_res = match Self::try_send(opts.clone(), datagram_conns, request).await {
-                Ok(response) if response.truncated() => {
-                    debug!("truncated response received, retrying over TCP");
-                    Ok(response)
-                }
-                Err(e) if opts.try_tcp_on_error || e.is_no_connections() => {
-                    debug!("error from UDP, retrying over TCP: {}", e);
-                    Err(e)
-                }
-                result => return result,
-            };
+            let udp_res: Result<DnsResponse, ProtoError> =
```

Comment:
Style nit: IMO it would be better to avoid type annotations in favor of more localized turbofishing or whatever is needed for the compiler.

Reply:
Fair, though in this case since it's a match statement, I find the compilation guarantee about the match arms results matching to be useful and makes it clearer.